### PR TITLE
Add cb58 Implementation and Tests

### PR DESCRIPTION
**Pull request by kylecbetts**
## Description

Added CB58 Class for encoding & decoding as well as tests for those functions.

## Testing instructions

Testing can be done through the test file Tests/CB58Test.cpp.

## Types of changes

New feature (non-breaking change which adds functionality)

## Checklist

- [x] Prefix PR title with `[WIP]` if necessary.
- [x] Add tests to cover changes as needed.
- [ ] Update documentation as needed.
- [x] I have read the [guidelines](https://developer.trustwallet.com/wallet-core/newblockchain#integration-criteria) for adding a new blockchain
- [x] If there is a related Issue, mention it in the description (e.g. Fixes #<issue_number> ).

### CODE REVIEW

**Comment by EvanJRichard** [comment]
Beautiful PR! Is there a way to merge our particular Avalanche `CB58` implementation with the ones in `Base58.h/Base58.cpp`?

**Comment by EvanJRichard** [comment]
In particular, I am thinking of how there are `public static` implementation objects in https://github.com/EvanJRichard/wallet-core/blob/master/src/Base58.h#L20

File: src/Hash.h [hunk 1]
```diff
@@ -170,6 +170,11 @@ Data groestl512(const T& data) {
     return groestl512(reinterpret_cast<const byte*>(data.data()), data.size());
 }
 
+/// Computes the SHA256 hash of the SHA256 hash.
```
Comment (EvanJRichard) [inline src/Hash.h]: Is this comment right for the new function?

File: src/CB58.cpp [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+
+#include "CB58.h"
+
+#include "Hash.h"
+
+#include <algorithm>
+#include <cctype>
+#include <cassert>
+
+using namespace TW;
```
Comment (EvanJRichard) [inline src/CB58.cpp]: Let's move this to namespace `TW::Avalanche` and therefore move the files into the `Avalanche` dir

File: src/CB58.h [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+#include "Data.h"
+#include "Hash.h"
+
+#include <array>
+#include <string>
+
+namespace TW {
+
+class CB58 {
```
Comment (EvanJRichard) [inline src/CB58.h]: Let's add a comment here explaining that the implementation is sufficiently different that we did not want to use `Base58`

File: src/CB58.cpp (hunk 2 above)
Comment (kylecbetts) [reply src/CB58.cpp]: I believe we are using the TW namespace in the .cpp file for calling some other functions such as Hashing. I think the correct thing to do is update the namespace declaration in CB58.h from TW to TW::Avalanche

**Review by EvanJRichard — APPROVED** [review]

**Pull request merged**